README.md

```diff
@@ -1,2 +1,107 @@
-# pystac-io
-Python library providing additional IO implementations for PySTAC
+# PySTAC IO
+
+PySTAC IO is a Python library that provides additional IO implementations for [`pystac.STAC_IO`](https://pystac.readthedocs.io/en/latest/concepts.html#using-stac-io).
+
+PySTAC supports reading and writing STAC catalogs via the local filesytem. The project provides:
+
+| Module | Scheme | Reads? | Writes? |
+|--------|------------|-------|--------|
+| https  | `https://` | X     |        |
+| s3     | `s3://`    | X     | X      |
+
+## Installation
+
+Clone then install locally into your environment via pip:
+
+```shell
+pip install .
+```
+
+You'll also need to install the dependencies for the modules you wish to use.
+For example, to use the `s3` module:  
+
+```shell
+pip install ".[s3]"
+```
+
+## Usage
+
+Import `pystac_io` and the submodule you plan to use, register the submodule to enable it, and then use `pystac` normally!
+
+```python
+import pystac
+import pystac_io
+import pystac_io.s3   # Import the IO module you plan to `register()`
+
+pystac_io.register("s3")
+s3_catalog = pystac.Catalog.from_file("s3://bucket/path/to/catalog.json")
+```
+
+### Advanced Usage
+
+#### Multiple IO Modules
+
+`pystac.STAC_IO` is only able to register a single global read and write handler. If you need to use multiple `pystac_io` modules in the same script you need to unregister one before registering another:
```

Comment:
This is screaming out for a context manager. I've never implemented one -- do you know if it's tricky?

```
with pystac_io.register("s3"):
    ...
```

seems like a much better idiom than requiring people to remember to unregister things.

Reply:
🔥 that's a great idea. Context managers are pretty straightforward, take a look at the [Examples](https://www.python.org/dev/peps/pep-0343/#id46) section of PEP343. Theres a function decorator so you don't have to construct a class with `__enter__` and `__exit__` methods.

Reply:
Ended up being a bit trickier than expected to get `pystac_io.register` to work as a regular function and be a context manager. But this is now updated.